examples/ex_palette.c

```diff
@@ -115,45 +94,30 @@ int main(int argc, char **argv)
       }
    }
 
+   al_set_new_bitmap_format(ALLEGRO_PIXEL_FORMAT_ANY);
+   pal_bitmap = al_create_bitmap(255, 7);
```

Comment:
OK, so I came across some very unexpected behavior here. Perhaps it is a bug? I'll explain.

I knew that allegro imposes a minimum texture size (16 for opengl. or maybe it is for any gfx backend?). So, originally I just had `16` here for the bitmap height, knowing I'd only use 7 rows and would need to divide by 15 in the fragment shader for the indexing (instead of 6, if the texture really were 7 pixels tall). All good, conceptually. But in practice, setting this value to `16` resulted in the entire screen being black.

I debugged via Spector.js and saw the texture in memory was padded in an unexpected way.

Here it is when you create the bitmap with height `7`:

<img width="597" alt="image" src="https://user-images.githubusercontent.com/4071474/156983739-fd7b3243-1197-4e7d-919e-19aa1fe17192.png">

And this is height `16`:

<img width="642" alt="image" src="https://user-images.githubusercontent.com/4071474/156983782-1a32217f-54dd-4b35-bb45-2588169674c1.png">

The bug is that I expected the additional padding that allegro would provide to go on the bottom and the right of the texture, leaving my "intended" texture in the top-left corner of the actual texture. The reality is this is reversed, resulting in the shader only ever getting black pixels (as the rest of the bitmap I'm not using it simply black).

I'm not sure if this misconception comes from my lack of familiarity with graphics pipelines (does padding in this manner make sense if you know more about how the system works?). And I'm sure changing this behavior is not a reasonable request. At least, it'd be nice if this behavior was documented [here](https://man.archlinux.org/man/community/allegro/al_create_bitmap.3.en#:~:text=If%20you%20try%20to%20create%20a%20bitmap%20smaller%20than%20this%2C%20this%20call%20will%20not%20fail%20but%20the%20returned%20bitmap%20will%20be%20a%20section%20of%20a%20larger%20bitmap%20with%20the%20minimum%20size.)

Reply:
I think what you're seeing here is that the texture is stored upside down. When the height is 7, it's padded on the bottom. When the height is 16 there's no padding, but the data is placed on the bottom because the texture is upside down.